api/src/main/java/org/openmrs/PatientIdentifier.java

```diff
@@ -262,6 +262,24 @@ public Integer getPatientIdentifierId() {
 	public void setPatientIdentifierId(Integer patientIdentifierId) {
 		this.patientIdentifierId = patientIdentifierId;
 	}
+
+	/**
+	 * bitwise copy of the PatientIdentifier object. NOTICE: THIS WILL NOT COPY THE PATIENT OBJECT. The
+	 * PatientIdentifier.patient object in this object AND the cloned object will point at the same
+	 * patient
+	 *
+	 * @return New PatientIdentifier object
+	 * @since 2.2.0
+	 */
+	@Override
+	public Object clone() {
+		try {
+			return super.clone();
```

Comment:
Yes i know we have done this for PersonAddress. Did you evaluate using a copy constructor? http://www.javapractices.com/topic/TopicAction.do?Id=71

Reply:
I did not evaluate using a copy constructor. Would adding the following code and making the object **Not** extend Clonable be better?

`
/**
	 * Constructor that takes in all fields to create a full identifier.
	 * 
	 * @param identifier
	 * @param type
	 * @param location
	 * @param patient
	 * @param preferred
	 * @param patientIdentifierId
	 */
	public PatientIdentifier(   String identifier,
								PatientIdentifierType type, 
								Location location,
								Patient patient,
								Boolean preferred,
								Integer patientIdentifierId
							) {
		this.identifier = identifier;
		this.identifierType = type;
		this.location = location;
		this.patient = patient;
		this.preferred = preferred;
		this.patientIdentifierId = patientIdentifierId;
	}

	/**
	 * Copy constructor that makes a shallow copy of the patient identifier passed in.
	 * Note that a copy of the patient, identifier type and location objects will not be created,
	 * the newly created patientIdentifier and the patientIdentifier that was passed in will reference the same objects.
	 * 
	 * @param patientIdentifier
	 */
	public PatientIdentifier(PatientIdentifier patientIdentifier) {
		this(   patientIdentifier.getIdentifier(),
				patientIdentifier.getIdentifierType(),
				patientIdentifier.getLocation(),
				patientIdentifier.getPatient(),
				patientIdentifier.getPreferred(),
				patientIdentifier.getPatientIdentifierId()
		); 
	}
`

Note that patientIdentifierType, patient and location are mutable objects I did not do a defensive copy (see http://www.javapractices.com/topic/TopicAction.do?Id=15) because these types don't have convent ways to create copies of them...? And if we aren't doing defensive copying, I wonder what is the difference between the clone and the copy constructor? what makes the clone so tricky to implement/dangerous?

Reply:
Fair enough. Merged. 😊